### PR TITLE
Bug 1668547: Incl. metric identifier in comment in gen. code

### PR DESCRIPTION
**Pull request by mdboom**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `HISTORY.rst` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean_parser/pings.py [hunk 1]
```diff
@@ -73,3 +73,9 @@ def serialize(self) -> Dict[str, util.JSONType]:
         d = self.__dict__.copy()
         del d["name"]
         return d
+
+    def identifier(self) -> str:
```
Comment (chutten) [inline glean_parser/pings.py:77]: pings.py? Shouldn't this be on Metric, not Ping?

**Review by chutten — COMMENTED** [review]
Going to add it to C# as well?

File: glean_parser/pings.py (hunk 1 above)
Comment (mdboom) [reply glean_parser/pings.py:77]: The same template is used to generate ping objects and metric objects, so we need `identifier` defined on both.  (Alternatively, could have put this logic in the templates, but that was much more complex...)

**Comment by mdboom** [comment]
> Going to add it to C# as well?

Yeah, I guess I might as well (even though that's on pause...)

File: HISTORY.rst [hunk 2]
```diff
@@ -5,6 +5,8 @@ History
 Unreleased
 ----------
 
+* Generated code now includes a comment next to each metric containing the name of the metric in its original `snake_case` form.
```
Comment (chutten) [inline HISTORY.rst:8]:
```suggestion
* Generated code now includes a comment next to each metric containing the name of the metric in its original `dotted.snake_case` form.
```

**Review by chutten — APPROVED** [review]

**Pull request merged**